Jenkinsfile

```diff
@@ -57,22 +67,32 @@ node('vetsgov-general-purpose') {
           if (commonStages.shouldBail()) { return }
           def envName = 'vagovstaging'
 
-          shouldBuild = !contentOnlyBuild || envName == params.cmsEnvBuildOverride
+          def shouldBuild = !contentOnlyBuild || envName == params.cmsEnvBuildOverride
           if (!shouldBuild) { return }
 
           try {
+            // Try to build using fresh drupal content
             commonStages.build(ref, dockerContainer, assetSource, envName, false, contentOnlyBuild)
             envUsedCache[envName] = false
           } catch (error) {
             if (!contentOnlyBuild) {
-              dockerContainer.inside(DOCKER_ARGS) {
-                sh "cd /application && node script/drupal-aws-cache.js --fetch --buildtype=${envName}"
+              try {
+                dockerContainer.inside(DOCKER_ARGS) {
+                  sh "cd /application && node script/drupal-aws-cache.js --fetch --buildtype=${envName}"
+                }
+                // Try to build again using cached drupal content
+                commonStages.build(ref, dockerContainer, assetSource, envName, true, contentOnlyBuild)
+                envUsedCache[envName] = true
+              } catch (err) {
+                throw err
               }
-              commonStages.build(ref, dockerContainer, assetSource, envName, true, contentOnlyBuild)
-              envUsedCache[envName] = true
             } else {
-              commonStages.build(ref, dockerContainer, assetSource, envName, false, contentOnlyBuild)
-              envUsedCache[envName] = false
+              try {
+                commonStages.build(ref, dockerContainer, assetSource, envName, false, contentOnlyBuild)
+                envUsedCache[envName] = false
+              } catch (err) {
+                throw err
+              }
```

Comment:
If you're only rethrowing the exception, how does it differ than letting it get thrown normally in the course of what's in the `try` block?

Reply:
Having just discovered the scoping mistakes that were causing unexpected retry behavior, I think that was the real culprit behind the issue that initiated this cleanup. It definitely helps that you're changing the `shouldBuild` variable to local scope.

Reply:
> If you're only rethrowing the exception, how does it differ from letting it get thrown normally in the course of what's in the `try` block?


I'm probably missing something but here's my thought process:

1. The [`build` function](https://github.com/department-of-veterans-affairs/vets-website/blob/c725abedf69abfb7d977bf55ae13e62db812f72b/jenkins/common.groovy#L245-L274) calls the [`checkForBrokenLinks` function](https://github.com/department-of-veterans-affairs/vets-website/blob/c725abedf69abfb7d977bf55ae13e62db812f72b/jenkins/common.groovy#L196-L243)
2. The [`checkForBrokenLinks` function can potentially `throw`](https://github.com/department-of-veterans-affairs/vets-website/blob/c725abedf69abfb7d977bf55ae13e62db812f72b/jenkins/common.groovy#L240).
3. The [initial `build` attempt for each env is wrapped in a `try` block](https://github.com/department-of-veterans-affairs/vets-website/blob/c725abedf69abfb7d977bf55ae13e62db812f72b/Jenkinsfile#L39-L42).
4. If the initial non-`contentOnlyBuild` attempt fails, then a [second build attempt will be made in the `catch` block using the drupal cache](https://github.com/department-of-veterans-affairs/vets-website/blob/c725abedf69abfb7d977bf55ae13e62db812f72b/Jenkinsfile#L44-L48). 
5. If the second attempt fails, I don't think we're currently handling that error.

Reply:
By "what's in the `try` block", I meant the new `try` block you introduced in the second attempt. My point is that re-throwing the exception isn't really doing anything with the error either, and it would probably behave the same as if you didn't try/catch in the second attempt.

Reply:
Ooooh. I think I see what you're saying. If I understand correctly, you're saying that my change is likely unnecessary because all I'm doing in the `catch` block is throwing the error, and I'm not really handling it, and therefore I'm not doing anything different than the default behavior, correct?

Reply:
I reverted my `try...catch` changes in 6ec565757245da9fc316355acbf7d985e82ab0c1. Now this draft PR is only updating variable scope, adding code comments, and deleting an unused function.

Reply:
Yup, that's what I was getting at. I'm not sure of the proper way to handle that issue where there was missing console output for the failed builds, but we'll have to monitor build failures and whether the scoped variables have eliminated the issue. 

I don't have enough of a sample size to determine the root cause, but from the one instance that you mentioned in the Slack thread, it seemed to run with `buildtype=vagovprod` with all the builds, which directly points to the previous variable scoping issue.